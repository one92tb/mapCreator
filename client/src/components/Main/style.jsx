import styled from "styled-components";

const RowStyle = styled.div`
  height: 100%;
  margin: 0 !important;
`;

const ColStyle = styled.div`
  padding: 0!important;
  height: 100%;

  @media only screen and (max-width: 991.98px) {
    height: auto;
        &:not(:first-child) {
          height: ${props =>
            props.currentLocation.pathname === "/" ? "100%" : "auto"};
          display: block;
        }
  }
`;

export { RowStyle, ColStyle };
