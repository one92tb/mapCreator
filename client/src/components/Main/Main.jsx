import React from "react";
import { Row, Col } from "reactstrap";
import { RowStyle, ColStyle } from "./style";
import { PanelComponent, Panel } from "./Panel/Panel";
import Map from "./Map/Map";
import MarkerCreator from "./MarkerCreator/MarkerCreator";
import { Switch, Route } from "react-router-dom";

Row.displayName = "div";
Col.displayName = "div";

const routes = [
  {
    path: "/",
    exact: true,
    section: Map
  },
  {
    path: "/createMarker",
    section: MarkerCreator
  }
];

export const Main = ({ ...props }) => (
  <Row tag={RowStyle}>
    <Col currentLocation={props.location} tag={ColStyle} lg="3" md="12">
      <PanelComponent {...props} />
    </Col>
    <Col currentLocation={props.location} tag={ColStyle} lg="9" md="12">
      <Switch>
        {routes.map((route, id) => (
          <Route
            key={id}
            path={route.path}
            exact={route.exact}
            component={route.section}
          />
        ))}
      </Switch>
    </Col>
  </Row>
);

export default Main;
