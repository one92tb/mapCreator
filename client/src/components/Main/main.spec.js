import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Main } from "./Main";

describe("main component", () => {
  it("shallow main component", () => {
    let wrapper;

    wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });
});
