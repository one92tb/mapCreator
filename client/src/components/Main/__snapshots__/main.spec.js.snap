// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main component shallow main component 1`] = `
<div
  tag={[Function]}
>
  <div
    lg="3"
    md="12"
    tag={[Function]}
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Connect(Panel) />
  </div>
  <div
    lg="9"
    md="12"
    tag={[Function]}
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        key="0"
        path="/"
      />
      <Route
        component={[Function]}
        key="1"
        path="/createMarker"
      />
    </Switch>
  </div>
</div>
`;
